### PR TITLE
GH-103903: Test the minimum Sphinx version in CI

### PR DESCRIPTION
**Pull request by AA-Turner**
* Issue: gh-103903

### CODE REVIEW

**Comment by AA-Turner** [comment]
@erlend-aasland -- were you just removing your own automatic review request, or did you mean to remove Alex & Cam too? The interface says we both edited at the same time so I wonder if there was clash -- just checking intentionality!

A

**Comment by AlexWaygood** [comment]
(I'm happy to review, but have just boarded the first of two flights back to the UK, so my review might be a little delayed! :)

**Comment by AA-Turner** [comment]
Safe journey!

**Comment by erlend-aasland** [comment]
I only meant to remove myself. I don't know anything about the various Sphinx versions, sorry. I'm sure both CAM and Alex are much more competent reviewers than me, for this PR.

File: .github/workflows/doc.yml [hunk 1]
```diff
@@ -105,3 +105,23 @@ jobs:
     # Use "xvfb-run" since some doctest tests open GUI windows
     - name: 'Run documentation doctest'
       run: xvfb-run make -C Doc/ PYTHON=../python SPHINXERRORHANDLING="-W --keep-going" doctest
+
+  # This build doesn't use problem matchers or check annotations
+  # It also does not run 'make check', as sphinx-lint is not installed into the
+  # environment.
+  build_doc_oldest_supported_sphinx:
```
Comment (hugovk) [inline .github/workflows/doc.yml]: Let's move this job between `build_doc` and `doctest`, easier to compare with the full `build_doc` one.

File: .github/workflows/doc.yml [hunk 2]
```diff
@@ -105,3 +105,23 @@ jobs:
...
+    name: 'Docs [Oldest supported Sphinx version]'
```
Comment (hugovk) [inline .github/workflows/doc.yml]: Let's keep this shorter to fit:

image

```suggestion
    name: 'Docs (oldest Sphinx)'
```

**Comment by hugovk** [comment]
> (I'm happy to review, but have just boarded the first of two flights back to the UK, so my review might be a little delayed! :)

Excuses, excuses! ;) Welcome home!

**Comment by AA-Turner** [comment]
I didn't expect the Spanish inquisition!

**Comment by AA-Turner** [comment]
Oh, does Sir Bedevere not recognise a regular 'comment' review?

A

**Review by hugovk — COMMENTED** [review]
> Oh, does Sir Bedevere not recognise a regular 'comment' review?

PR looks good now, but just testing this out... Want to try it again?

**Comment by AA-Turner** [comment]
I didn't expect the Spanish inquisition!

**Comment by AA-Turner** [comment]
Same result -- I'd guess that he picks up the key-phrase but doesn't recognise who to ping for a re-review?

A

**Review by AlexWaygood — APPROVED** [review]
This LGTM! Thanks @AA-Turner

**Comment by AlexWaygood** [comment]
@AA-Turner, can I leave the backport in your capable hands? :)

**Comment by AA-Turner** [comment]
> @AA-Turner, can I leave the backport in your capable hands? :)

Yessir, just having dinner though -- will do later this eve!

A

**Pull request merged**